### PR TITLE
Remove dependencies on glog

### PR DESCRIPTION
**Pull request by thaink**
This is an effort to make this branch buildable for Android.
- Remove all deps on glog, replacing it by the log implementation on master branch
- Refactor the deps on gflags in src/BUILD

### CODE REVIEW

**Comment by thaink** [comment]
@taku910 @resec Could you take a look at this PR?

**Comment by thaink** [comment]
Also update protobuf to v3.9.2 since v3.6.1.2 is deprecated.

**Comment by thaink** [comment]
PTAL @taku910

File: src/common.h [hunk 1]
```diff
@@ -16,15 +16,111 @@
...
+}
+}  // namespace logging
+}  // namespace sentencepiece
+
+#define LOG(severity)                                                     \
+  ::sentencepiece::error::Die(::sentencepiece::logging::LOG_##severity >= \
+                              ::sentencepiece::logging::LOG_FATAL) &      \
+      std::cerr << ::sentencepiece::logging::BaseName(__FILE__) << "("    \
+                << __LINE__ << ") "                                       \
+                << "LOG(" << #severity << ") "
```
Comment (broken) [inline src/common.h:87]: should this order be reversed? I believe bitwise-and reads left to right, so the newline will occur before the message.

File: src/common.h [hunk 2]
```diff
@@ -16,15 +16,111 @@
...
+
+#define CHECK(condition)                                                      \
+  (condition) ? 0                                                             \
+              : ::sentencepiece::error::Die(true) &                           \
+                    std::cerr << ::sentencepiece::logging::BaseName(__FILE__) \
+                              << "(" << __LINE__ << ") [" << #condition       \
+                              << "] "
```
Comment (broken) [inline src/common.h:94]: Here too. Will the program exit before the message is printed?

**Review by broken — COMMENTED** [review]
thanks for making this update!

File: src/common.h (hunk 1 above)
Comment (thaink) [reply src/common.h:87]: It is fine as Abort only get called in destructor.
The '&' operator ensure the log get printed before that.

File: src/common.h (hunk 2 above)
Comment (thaink) [reply src/common.h:94]: Same as above.

File: src/common.h (hunk 1 above)
Comment (broken) [reply src/common.h:87]: Ah; yes. I completely overlooked that. Thanks.

**Comment by taku910** [comment]
Just a confirmation: Is this PR for bazel branch?  Don't we have any side-effect on master branch?

**Comment by thaink** [comment]
Right. This PR is only for the bazel branch. It will not affect the master branch.

**Comment by thaink** [comment]
Do you think the continuous test fail is related to this PR?

**Comment by thaink** [comment]
Can I have a LGTM?

**Comment by taku910** [comment]
merged. Thank you

**Pull request merged**